[CARBONDATA-3676] Support clean carbon data files of stages. (#3602)
apache/carbondata: docs/dml-of-carbondata.md

  ```
  DELETE FROM carbontable WHERE column1 IN (SELECT column11 FROM sourceTable2 WHERE column1 = 'USA')
  ```
    

> ### DELETE STAGE

Reviewer: ```suggestion
### DELETE STAGE files
```

> ### DELETE STAGE

Reviewer 2: Can you please add documentation for flink integration to carbon and can move these new DDL's added for flink and stage to it.

Author: No DDL's added  for flink.
Both CarbonInsertFromStageCommand and CarbonDeleteStageFilesCommand are DML.

Reviewer 2: okay.. Can you please add documentation for flink integration to carbon and stage files

Author: OK

>   This command allows us to delete data files which referenced by disabled table stages.

Reviewer: ```suggestion
  This command allows us to delete the data files (stage data) which is already loaded into the table.
```

Author: Accept suggestion, and updated.

  ```
  DELETE FROM TABLE [db_name.]table_name STAGE OPTIONS(property_name=property_value, ...)
  ```  
  **Supported Properties:**

| Property                                                     | Description                                                  |
| ------------------------------------------------------------ | ------------------------------------------------------------ |

> | [data_file_location](#data_file_location)                    | The data files location                                      |

Reviewer: ```suggestion
| [location](#data_file_location)                    | The data file location                                      |
```

Author: Accept suggestion, and updated.

> | [data_file_retain_time_second](#data_file_retain_time_second)| Data file retain time in second                              |

Reviewer: ```suggestion
| [retain_hour](#retain_hour)| Data file retain time in hours                              |
```

Author: Accept suggestion, and updated.

-
  You can use the following options to delete data:

  - ##### data_file_location: 

>     The data files location, the command will scan the location, and delete files which can be deleted.

Reviewer: ```suggestion
    The data files location, the command will scan the location, and delete files which are already loaded into the table. This check is done by scanning the stage metadata folder in table path.
```

Author: Accept suggestion, and updated.
